HHS/Head-Start-TTADP
# [TTAHUB-2317] AR groups

## Description of change

When creating a recipient AR we now allow the user to select a group that will pre fill the list of recipients. If the pre filled list of recipients is altered in anyway (add/remove). The user is informed that they have modified the group recipients and will be given the option to start over.

**NOTE 12/07/2023:** Right now we are getting all groups by region regardless of shared users as this change is still pending on the BE. **ATM the FE can be reviewed.**

## How to test

- Review all code.
- Create a recipient report and test the workflow from the UX Workflow:
 https://www.figma.com/proto/zciEyTLd2MeSK7z5YVKdbF/Create-AR-and-TR-session-from-%22my-group%22?page-id=1%3A52&type=design&node-id=531-7254&viewport=-848%2C-3704%2C0.25&t=GBVp5tULq3N2QVO3-1&scaling=min-zoom&starting-point-node-id=531%3A7254&show-proto-sidebar=1&mode=design*
- Test validation for groups.
- Make sure the group displayed are correct for the region and number of recipients populated.
- Test edge cases of selecting and removing groups also having pre selected recipients and switching (and vice versa).
- Make sure a report can be created and approved successfully when using and not using groups.




## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2317


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
